osf/migrations/0170_populate_file_versions_through.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-03-03 17:52
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations, connection
+
+logger = logging.getLogger(__file__)
+
+
+def restore_default_through_table(state, schema):
+    sql = """
+        DROP TABLE osf_basefilenode_versions;
+        CREATE TABLE osf_basefilenode_versions AS
+        SELECT
+            new_thru.basefilenode_id,
+            new_thru.fileversion_id
+        FROM
+            osf_basefileversionsthrough AS new_thru;
+
+        ALTER TABLE osf_basefilenode_versions ADD COLUMN id SERIAL PRIMARY KEY;
+        ALTER TABLE osf_basefilenode_versions ADD CONSTRAINT osf_basefilenod_basefilenode_id_b0knah27_fk_osf_basefilenode_id FOREIGN KEY (basefilenode_id) REFERENCES osf_basefilenode DEFERRABLE INITIALLY DEFERRED;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN basefilenode_id
+        SET
+            DATA TYPE INTEGER;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN fileversion_id
+        SET
+            NOT NULL;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN fileversion_id
+        SET
+            DATA TYPE INTEGER;
+        ALTER TABLE osf_basefilenode_versions ALTER COLUMN basefilenode_id
+        SET
+            NOT NULL;
+        ALTER TABLE osf_basefilenode_versions ADD CONSTRAINT osf_basefilenode__fileversion_id_93etanfc_fk_osf_fileversion_id FOREIGN KEY (fileversion_id) REFERENCES osf_fileversion DEFERRABLE INITIALLY DEFERRED;
+        ALTER TABLE osf_basefilenode_versions ADD CONSTRAINT osf_basefilenode__fileversion_uniq564 UNIQUE (basefilenode_id, fileversion_id);
+        CREATE INDEX
+        ON osf_basefilenode_versions (basefilenode_id, fileversion_id);
+        CREATE INDEX
+        ON osf_basefilenode_versions (basefilenode_id);
+        CREATE INDEX
+        ON osf_basefilenode_versions (fileversion_id);
+    """
+    with connection.cursor() as cursor:
+        cursor.execute(sql)
+
+
+def populate_fileversion_name(state, schema):
+
+    sql = """
+        DROP TABLE osf_basefileversionsthrough;
+        CREATE TABLE osf_basefileversionsthrough AS
+        SELECT
+            obfv.basefilenode_id,
+            obfv.fileversion_id,
+            ob.name as version_name
```

Comment:
Another thing - how thorough do we need to be with this - this just adds the current file name to all the versions.  I have file Dog - v1 was named Cat, v2 was renamed to Dog.  In this migration, v1 and v2 will both be named Dog.  (For all intents and purposes, all downloads of v1 to date have been named Dog).  I could potentially explore digging through the logs to get these names?   I thought I remembered a conversation with someone where the consensus was "don't worry about it" for these older ones, because we weren't intentionally tracking it before.  But worth asking -

Reply:
No, no, don't go crazy with that. As far as versions are concerned up until this point, renaming the file renames the versions. I will double check with Product, but I'm pretty sure they won't want to spend time on that. If anything, they'd just make a note in the help guides about how this got better or something, so for files going forward, you don't *have* to dig through the logs to see how the filename changed.

Reply:
ok, great